### ISSUE TITLE
Incorrect use of torchaudio's rnnt_loss

### ISSUE THREAD

**Issue by csukuangfj** [speechbrain/speechbrain#1339]
It is suprising that no one has noticed that the current code is feeding the output of 
log_softmax to torchaudio's rnnt_loss, which is not correct.

torchaudio's rnnt_loss accepts only `logits` as input.

I am wondering whether anyone has succeeded in training a transducer model
with the current usage of torchaudio's rnnt_loss using speechbrain.

---

The relevant code is listed below:

https://github.com/speechbrain/speechbrain/blob/24720c1532355dfeeda78304939dcc80283244bf/recipes/CommonVoice/ASR/transducer/train.py#L75

https://github.com/speechbrain/speechbrain/blob/24720c1532355dfeeda78304939dcc80283244bf/recipes/CommonVoice/ASR/transducer/train.py#L136-L138

https://github.com/speechbrain/speechbrain/blob/24720c1532355dfeeda78304939dcc80283244bf/recipes/CommonVoice/ASR/transducer/hparams/train_fr.yaml#L183-L185

https://github.com/speechbrain/speechbrain/blob/24720c1532355dfeeda78304939dcc80283244bf/speechbrain/nnet/losses.py#L59-L77

**Comment by vsokolov00** [comment]
I'm trying to train an RNN-T model... Have you tried to train a model with this error fixed? How much was the difference?

**Comment by csukuangfj** [comment]
No, I have not used it to train a model. I just spotted this bug while looking at the code.

**Comment by aheba** [comment]
Hello @csukuangfj , thank you for reporting the issue, 
let me do a fix

**Comment by vsokolov00** [comment]
It's strange but I didn't notice any improvements during the training process, after fixing this bug it produces even worse results during the validation.. I'll make the update once I decode it on the test set

**Comment by vsokolov00** [comment]
It's strange but I didn't notice any improvements during the training process, after fixing this bug it produces even worse results during the validation.. I'll make the update once I decode it on the test set

**Comment by vsokolov00** [comment]
softmax outputs - `Epoch loaded: 30 - test loss: 31.53, test WER: 55.05, test CER: 41.22`
(FIXED) logits - `Epoch` loaded: 30 - test loss: 30.48, test WER: 54.91, test CER: 41.20`

**Comment by Adel-Moumen** [comment]
Hello, 

This issue has been solved with #1368, so I'm closing this issue. 
Thanks again for reporting this problem! :-)
